sites/hashdev/src/_pages/blog/5_blocks-without-typescript.mdx

```diff
@@ -0,0 +1,509 @@
+---
+authorPhoto: blog/5_blocks-without-typescript/ahmad.png
+postPhoto: blog/3_one-principle/3_one-principle.jpg
+title: "Building Block Protocol blocks without TypeScript"
+subtitle: ""
+author: "Ahmad Sattar Atta"
+jobTitle: "Platform Engineer at HASH"
+date: "2022-06-10"
+---
+
+Version 0.2 of the Block Protocol is hot off the press!
+With this new version interactions between blocks and Embedding Applications have changed significantly in order to support blocks written using a wider variety of languages and technologies.
+Block Protocol requests and responses are to be handled through DOM `CustomEvent`s defined according to backing JSON Schemas.
+
+This means that it is now easier than ever to create blocks that are not tied to JavaScript/TypeScript and/or React.
+As long as a block can talk browser `Event`s, it can implement the Block Protocol.
+To show what the changes imply for adventurous block authors, this post will be a technical walk-through of how the Block Protocol can be implemented and used in a language that is not TypeScript.
+To accomplish this, we'll go through how a block can be developed in F# — but the methodology and implementation details shown are applicable for other language that supports JavaScript as a transpilation target and JavaScript interoperability.
+
+## What are we going to build?
+
+[F#](https://fsharp.org/) along with [Fable compiler](https://fable.io/) and [Feliz](https://zaid-ajaj.github.io/Feliz/) is a great way to develop for the web if functional programming is of interest.
+Here it will be used to create a logically complicated _calculation block_ resembling a spreadsheet that allows calculation, aggregation and data sourcing from Embedding Applications.
+
+The purpose of the Block Protocol from the perspective of this calculation block is to be an interface for any Embedding Application.
+This interface can be thought of as a data provider/consumer layer: a sort of backend for the block, and a sort of frontend for the embedding application. Writing our calculation block _once_ with the Block Protocol allows any BP-compliant Embedding Application to be a possible data provider and make use of it.
+
+<CalculationBlock></CalculationBlock>
```

Comment:
I think it would be good to make it explicit that the block is actually the block and not an image. It looks too much like an image by itself.